delete_version_after_test.go

```diff
@@ -0,0 +1,265 @@
+	t.Helper()
+	return getTime(t, "deletion_time", d).Sub(ut)
+}
+
+func lifetime(t *testing.T, d map[string]interface{}) time.Duration {
+	t.Helper()
+	ct := getTime(t, "created_time", d)
+	return untilDeletion(t, ct, d)
+}
+
+func TestDeleteVersionAfter(t *testing.T) {
+	nd := -1 * time.Second
+	dm, ds := 6*time.Hour, 3*time.Hour
+	var tests = []struct {
+		mount, meta      time.Duration
+		want             time.Duration
+		wantDeletionTime bool
+	}{
+		{0, 0, 0, false},
+		{0, ds, ds, true},
+		{ds, 0, ds, true},
+		{dm, ds, ds, true},
+		{ds, dm, ds, true},
+		{nd, 0, 0, false},
+		{nd, ds, 0, false},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(fmt.Sprintf("mount=%v,meta=%v", tt.mount, tt.meta), func(t *testing.T) {
+			t.Parallel()
+
+			b, storage := getBackend(t)
+
+			data := map[string]interface{}{
+				"delete_version_after": tt.mount.String(),
+			}
+			req := &logical.Request{
+				Operation: logical.CreateOperation,
+				Path:      "config",
+				Storage:   storage,
+				Data:      data,
+			}
+			resp, err := b.HandleRequest(context.Background(), req)
+			wantNoResponse(t, resp, err)
+
+			req = &logical.Request{
+				Operation: logical.ReadOperation,
+				Path:      "config",
+				Storage:   storage,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantResponse(t, resp, err)
+			if tt.mount == 0 {
+				got := resp.Data["delete_version_after"]
+				if got != nil {
+					t.Fatalf("mount value: delete_version_after %#v, want no delete_version_after", got)
+				}
+			} else {
+				want, got := tt.mount.String(), resp.Data["delete_version_after"]
+				if want != got {
+					t.Fatalf("want delete_version_after: %v, got %v", want, got)
+				}
+			}
+
+			data = map[string]interface{}{
+				"max_versions":         2,
+				"delete_version_after": tt.meta.String(),
+			}
+			req = &logical.Request{
+				Operation: logical.CreateOperation,
+				Path:      "metadata/foo",
+				Storage:   storage,
+				Data:      data,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantNoResponse(t, resp, err)
+
+			req = &logical.Request{
+				Operation: logical.ReadOperation,
+				Path:      "metadata/foo",
+				Storage:   storage,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantResponse(t, resp, err)
+			want, got := tt.meta.String(), resp.Data["delete_version_after"]
+			if want != got {
+				t.Fatalf("want delete_version_after: %v, got %v", want, got)
+			}
+
+			data = map[string]interface{}{
+				"data": map[string]interface{}{
+					"bar": "baz1",
+				},
+			}
+			req = &logical.Request{
+				Operation: logical.CreateOperation,
+				Path:      "data/foo",
+				Storage:   storage,
+				Data:      data,
+			}
+			resp, err = b.HandleRequest(context.Background(), req)
+			wantResponse(t, resp, err)
+			if !tt.wantDeletionTime {
+				if dtv := resp.Data["deletion_time"].(string); dtv != "" {
+					t.Logf("resp: %#v", resp)
```

Comment:
Did you mean to leave this in?

Reply:
Yes, because if there is a `deletion_time` it is an error so seeing what the `deletion_time` is will probably help identify the problem. However, the one on line 256 shouldn't be there, it should be after line 252. I will move it.